geonetwork/content.md

```diff
@@ -12,50 +12,130 @@ The project is part of the Open Source Geospatial Foundation ( [OSGeo](http://ww
+
+```yaml
+services:
+  geonetwork:
+    image: geonetwork:4.4.0
+    environment:
+      GN_CONFIG_PROPERTIES: >-
+        -Des.host=elasticsearch
+        -Des.protocol=http
+        -Des.port=9200
+        -Des.url=http://elasticsearch:9200
+        -Des.username=my_es_username
+        -Des.password=my_es_password
+        -Dkb.url=http://kibana:5601
+```
+
+#### Environment variables (version older than 4.4.0)
+
+If you are using a GeoNetwork older than 4.4.0 it must be configured using these environment variables:
+
+-	`ES_HOST` **mandatory**: The host name of the Elasticsearch server.
+-	`ES_PORT` *optional* (default `9200`): The port where Elasticsearch server is listening to.
+-	`ES_PROTOCOL` *optional* (default `http`): The protocol used to talk to Elasticsearch. Can be `http` or `https`.
+-	`ES_INDEX_RECORDS` *optional* (default `gn_records`): In case you have more than GeoNetwork instance using the same Elasticsearch cluster each one needs to use a different index name. Use this variable to define the name of the index used by each GeoNetwork.
+-	`ES_USERNAME` *optional (default empty): username used to connect to Elasticsearch.
+-	`ES_PASSWORD` *optional (default empty): password used to connect to Elasticsearch.
+-	`KB_URL` *Optional* (default `http://localhost:5601`): The URL where Kibana is listening.
+
+### Database configuration
+
+By default GeoNetwork uses a local H2 database but this one is not recommended for production use. The image contains JDBC drivers for PostgreSQL and MySQL. To configure the database connection use these environment variables:
 
-## Start geonetwork
+-	`GEONETWORK_DB_TYPE`: The type of database to use. Valid values are `postgres`, `postgres-postgis`, `mysql`. The image can be extended including other drivers and these other types could be used too: `db2`, `h2`, `oracle`, `sqlserver`. The JAR drivers for these other databases would need to be added to `/opt/geonetwork/WEB-INF/lib` mounting them as binds or extending the official image.
+-	`GEONETWORK_DB_HOST`: The database host name.
+-	`GEONETWORK_DB_PORT`: The database port.
+-	`GEONETWORK_DB_NAME`: The database name.
+-	`GEONETWORK_DB_USERNAME`: The username used to connect to the database.
+-	`GEONETWORK_DB_PASSWORD`: The password used to connect to the database.
+-	`GEONETWORK_DB_CONNECTION_PROPERTIES`: Additional properties to be added to the connection string, for example `search_path=test,public&ssl=true` will produce a JDBC connection string like `jdbc:postgresql://localhost:5432/postgres?search_path=test,public&ssl=true`
 
-This command will start a debian-based container, running a Tomcat web server, with a geonetwork war deployed on the server:
+### Start geonetwork
+
+This command will start a debian-based container, running a Tomcat (GN 3) or Jetty (GN 4) web server, with a geonetwork war deployed on the server:
 
 ```console
-$ docker run --name some-%%REPO%% -d %%IMAGE%%
+docker run --name some-%%REPO%% -d %%IMAGE%%
```

Comment:
Thanks for removing the `$` is allows the command to be cut and pasted.

What is missing here is instructions on starting Elasticsearch. When people sat down at foss4g-asia code sprint they wanted to know exactly what to run that would work.

Given that GeoNetwork 4.4.0 requires a specific version range of Elasticsearch in order to operate I think it would be a good thing to have in the example.

Looking at https://docs.geonetwork-opensource.org/4.4/install-guide/installing-index/ the version is 7.17.15 - but no instructions are provided for running Docker.

Looking at developer instructions https://github.com/geonetwork/core-geonetwork/blob/main/es/README.md it has:
> docker pull docker.elastic.co/elasticsearch/elasticsearch:7.6.2
> docker run -p 9200:9200 -p 9300:9300 -e "discovery.type=single-node" docker.elastic.co/elasticsearch/elasticsearch:7.6.2

This is the information missing from this file.

Reply:
I'd prefer that people use the compose file `stack.yml` included in this README. When this doc is built the file is shown inline.

Reply:
The we should list that first, if it is the easiest approach.

Reply:
Updated with instructions about how to run GN with ES in this section.